doc/tut2.rst

```diff
@@ -603,19 +603,22 @@ Turning the ``log`` proc into a template solves this problem:
     x = 4
   log("x has the value: " & $x)
 
-The parameters' types can be ordinary types or the meta types ``expr``
-(stands for *expression*), ``stmt`` (stands for *statement*) or ``typedesc``
-(stands for *type description*). If the template has no explicit return type,
-``stmt`` is used for consistency with procs and methods.
+The parameters' types can be ordinary types or the meta types ``untyped``,
+``typed``, or ``typedesc``.
+``typedesc`` stands for *type description*, and ``untyped`` means symbol lookups and
+type resolution is not performed before the expression is passed to the template.
 
-If there is a ``stmt`` parameter it should be the last in the template
-declaration. The reason is that statements can be passed to a template
+If the template has no explicit return type,
+``void`` is used for consistency with procs and methods.
+
+If there is a ``typed`` parameter, it should be the last in the template
```

Comment:
@Araq, I was also thinking of also adding (below the example) the following


Note: ``txt`` is a parameter to the withFile template that doesn't require 
explicit definition prior to being passed to withFile().
``txt`` is defined within withFile(), but needs to be a parameter so it can 
be used in the block of statements that follow withFile()